cds_ils/migrator/api.py

```diff
@@ -250,15 +264,232 @@ def import_items_from_json(dump_file, include, rectype="item"):
+            else:
+                client_id = current_app.config["CERN_APP_OPENID_CREDENTIALS"][
+                    "consumer_key"
+                ]
+                account = RemoteAccount.get(
+                    user_id=user.id, client_id=client_id
+                )
+                extra_data = account.extra_data
+                # add legacy_id information
+                account.extra_data.update(legacy_id=record["id"], **extra_data)
+                db.session.add(account)
+                patron = Patron(user.id)
+                PatronIndexer().index(patron)
+        db.session.commit()
+
+
+def import_loans_from_json(dump_file):
+    """Imports loan objects from JSON."""
+    dump_file = dump_file[0]
+    loans = []
+    with click.progressbar(json.load(dump_file)) as bar:
+        for record in bar:
+            click.echo('Importing loan "{0}"...'.format(record["legacy_id"]))
+            user = get_user_by_legacy_id(record["id_crcBORROWER"])
+            if not user:
+                patron_pid = SystemAgent.id
+            else:
+                patron_pid = user.pid
+            try:
+                item = get_item_by_barcode(record["item_barcode"])
+            except ItemMigrationError:
+                continue
+                # Todo uncomment when more data
+                # raise LoanMigrationError(
+                #    'no item found with the barcode {} for loan {}'.format(
+                #        record['item_barcode'], record['legacy_id']))
+
+            # additional check if the loan refers to the same document
+            # as it is already attached to the item
+            document_pid = item.get("document_pid")
+            document = Document.get_record_by_pid(document_pid)
+            if record["legacy_document_id"] is None:
+                raise LoanMigrationError(
+                    "no document id for loan {}".format(record["legacy_id"])
+                )
+            if (
+                document.get("legacy_recid", None)
+                != record["legacy_document_id"]
+            ):
+                # this might happen when record merged or migrated,
+                # the already migrated document should take precedence
+                click.secho(
+                    "inconsistent document dependencies for loan {}".format(
+                        record["legacy_id"]
+                    ),
+                    fg="blue",
+                )
+
+            # create a loan
+            (
+                default_location_pid_value,
+                _,
+            ) = current_app_ils.get_default_location_pid
+
+            if record["status"] == "on loan":
+                loan_dict = dict(
+                    patron_pid=str(patron_pid),
+                    transaction_location_pid=default_location_pid_value,
+                    transaction_user_pid=str(SystemAgent.id),
+                    document_pid=document_pid,
+                    item_pid={
```

Comment:
what about just:
```suggestion
                    item_pid=item.pid
```
?